File: C++/Basic Hashing/StringHashing.cpp
#include <iostream>
using namespace std;

int main() 
{
 
   string str;
   std::cin >> str;
   int q;
   cin>>q;
   int hash[str.size()+1]={0};
   
   for(int i=0;i<str.size();i++){
     hash[str[i]-'a']+=1;
   }
   
   while(q--){
     char c;
     cin>>c;
     std::cout << hash[c-'a'] << std::endl;
   }
   
    return 0;
}

File: Javascript/Basic Hashing/StringHashing.js



//basic hashing in javascript;

// what is hashing it's just pre compute and fetching method


//Hashing in string;

function main(){
  
    let str="abcdeae";
    let q=2
    let s=['a','e'];
    // filling the array with zero which is bigger than the size of string
    let hash=new Array(str.length+1).fill(0);
    let c='a'
     for(let i=0;i<str.length;i++){
      hash[str[i].charCodeAt() - c.charCodeAt()]+=1;
     }
     
    
    while(q--){
        // number of queue charcter checking
       console.log(hash[s[q].charCodeAt()-c.charCodeAt()])
    }
    
    
   
    
   
    
    
  }
  
  main()